### DISCUSSION TITLE
Unable to pass query string parameters to .NET Core Web API running in Lambda

### DISCUSSION THREAD

**Discussion by sudarshankonge** [aws/aws-lambda-dotnet#1166] [Q&A] [answered]
### Describe the bug

1. I have a Lambda function which is running a .NET Core Web API app from a container image. I am using .NET 6 for the Lambda.
2. I am using Lambda proxy integration with API gateway.
3. One of my APIs takes two query string parameters named `start` and `end`,
![image](https://user-images.githubusercontent.com/97505647/166007327-4eb97c7f-4241-4827-9d35-c1f318edc1a9.png)

4. I made the requisite changes so that API gateway can accept query string parameters as shown below,
![image](https://user-images.githubusercontent.com/97505647/166007051-8a7ca714-d289-4bf7-8bad-a477f9558f95.png)

When I try to invoke the above API from Postman, **I get a 404 not found error.** This is how the API looks,
`GET https://xxxxxxx-api.ap-south-1.amazonaws.com/dev/api/v1/stock-exchange/nse/time-series/INFY?start=2022-04-26&end=2022-04-30`

Can someone please help me debug this. From the logs, I can clearly see that the query string params are there in the request that Lambda gets but why is **my controller action failing to be invoked** ?

### Expected Behavior

WebAPI controller action must be invoked when I execute the API with the query string parameters.

### Current Behavior
This is the output I get after setting `LAMBDA_NET_SERIALIZER_DEBUG` to true.
```
{
    "resource": "/{proxy+}",
    "path": "/api/v1/stock-exchange/nse/time-series/INFY",
    "httpMethod": "GET",
    "headers": {
        "Accept": "*/*",
        "Accept-Encoding": "gzip, deflate, br",
        "Host": "xxxxx-api.ap-south-1.amazonaws.com",
        "Postman-Token": "65ae1a29-803a-4c17-8ede-6c02fa0c4578",
        "User-Agent": "PostmanRuntime/7.29.0",
        "X-Amzn-Trace-Id": "Root=1-626c2e55-707affc125a142870c4febb9",
        "X-Forwarded-For": "106.201.80.99",
        "X-Forwarded-Port": "443",
        "X-Forwarded-Proto": "https"
    },
    "multiValueHeaders": {
        "Accept": [
            "*/*"
        ],
        "Accept-Encoding": [
            "gzip, deflate, br"
        ],
        "Host": [
            "xxxxxxxx-api.ap-south-1.amazonaws.com"
        ],
        "Postman-Token": [
            "65ae1a29-803a-4c17-8ede-6c02fa0c4578"
        ],
        "User-Agent": [
            "PostmanRuntime/7.29.0"
        ],
        "X-Amzn-Trace-Id": [
            "Root=1-626c2e55-707affc125a142870c4febb9"
        ],
        "X-Forwarded-For": [
            "106.201.80.99"
        ],
        "X-Forwarded-Port": [
            "443"
        ],
        "X-Forwarded-Proto": [
            "https"
        ]
    },
    "queryStringParameters": {
        "end": "2022-04-30",
        "start": "2022-04-26"
    },
    "multiValueQueryStringParameters": {
        "end": [
            "2022-04-30"
        ],
        "start": [
            "2022-04-26"
        ]
    },
    "pathParameters": {
        "proxy": "api/v1/stock-exchange/nse/time-series/INFY"
    },
    "stageVariables": null,
    "requestContext": {
        "resourceId": "xxxxxxx",
        "resourcePath": "/{proxy+}",
        "httpMethod": "GET",
        "extendedRequestId": "RWwtdF9OBcwFuTQ=",
        "requestTime": "29/Apr/2022:18:28:37 +0000",
        "path": "/dev/api/v1/stock-exchange/nse/time-series/INFY",
        "accountId": "xxxxxxxxxx",
        "protocol": "HTTP/1.1",
        "stage": "dev",
        "domainPrefix": "gs9bhhllvh",
        "requestTimeEpoch": 1651256917914,
        "requestId": "0ed0b37f-bfd1-4c26-b18c-9313388ad4b8",
        "identity": {
            "cognitoIdentityPoolId": null,
            "accountId": null,
            "cognitoIdentityId": null,
            "caller": null,
            "sourceIp": "106.201.80.99",
            "principalOrgId": null,
            "accessKey": null,
            "cognitoAuthenticationType": null,
            "cognitoAuthenticationProvider": null,
            "userArn": null,
            "userAgent": "PostmanRuntime/7.29.0",
            "user": null
        },
        "domainName": "xxxxxxxxxxx-api.ap-south-1.amazonaws.com",
        "apiId": "xxxxxxxxxx"
    },
    "body": null,
    "isBase64Encoded": false
}


```

### Reproduction Steps

`GET https://xxxxxxx-api.ap-south-1.amazonaws.com/dev/api/v1/stock-exchange/nse/time-series/INFY?start=2022-04-26&end=2022-04-30`

![image](https://user-images.githubusercontent.com/97505647/166008663-1a3b2188-7686-4427-ae55-130d0914d887.png)

Executing the above API should invoke the WebAPI controller above but that doesn't happen and I get a 404 not found instead.

### AWS .NET SDK and/or Package version used

AWSSDK 2.3.55.2

### Targeted .NET Platform

.NET 6

### Operating System and version

Lambda

**Comment by ashishdhingra** [comment]
@sudarshankonge Could you try the URL https://xxxxxxx-api.ap-south-1.amazonaws.com/Dev/api/v1/stock-exchange/nse/time-series/INFY?start=2022-04-26&end=2022-04-30 (notice capital `D` in `Dev`) and see if it works? If this doesn't work, please share the sample code to troubleshoot the issue.

**Comment by ashishdhingra** [comment, marked as answer]
@sudarshankonge Somehow I'm unable to reproduce the issue. Here are the brief steps used for testing:
1. Using Visual Studio 2022, created a Serverless WebAPI project that targets .NET 6.0.
2. Replaced the default generated `ValuesController` with the following code:
```C#
using Microsoft.AspNetCore.Mvc;

namespace LambdaWebAPINet6_Issue1164.Controllers;

[Route("api/[controller]")]
public class ValuesController : ControllerBase
{
    [ProducesResponseType(typeof(TimeSeriesResponse), StatusCodes.Status200OK)]
    [HttpGet("/api/v1/stock-exchange/{stockExchange}/time-series/{stockExchangeSymbol}")]
    public ActionResult<TimeSeriesResponse> GetTimeSeries(
        [FromRoute] string stockExchange,
        [FromRoute] string stockExchangeSymbol,
        [FromQuery] DateTime start,
        [FromQuery] DateTime end)
    {
        return new TimeSeriesResponse() { StockExchange = stockExchange, StockExchangeSymbol = stockExchangeSymbol, Status = "OK" };
    }

    public class TimeSeriesResponse
    {
        public string? StockExchange { get; set; }

        public string? StockExchangeSymbol { get; set; }

        public string? Status { get; set; }
    }
}
```
4. Published the project to AWS Lambda. Default publishing process generates `Prod` and `Stage` stages. 
5. Tested the URL `https://<<API_ID>>.execute-api.us-east-2.amazonaws.com/Prod/api/v1/stock-exchange/nse/time-series/INFY?start=2022-04-26&end=2022-04-30`. It generated the following response:
<img width="1128" alt="Screen Shot 2022-04-29 at 4 24 01 PM" src="https://user-images.githubusercontent.com/67916761/166080313-35fa9096-4e5d-4d92-99e8-b924eda12811.png">


6. Defined another stage `Dev` from AWS console. The URL `https://<<API_ID>>.execute-api.us-east-2.amazonaws.com/Dev/api/v1/stock-exchange/nse/time-series/INFY?start=2022-04-26&end=2022-04-30` also generated the same response.

I didn't had to define query string parameters in API gateway settings. Kindly note that the stage names are case sensitive. So the URL `https://<<API_ID>>.execute-api.us-east-2.amazonaws.com/dev/api/v1/stock-exchange/nse/time-series/INFY?start=2022-04-26&end=2022-04-30` gives `Forbidden` error.

Thanks,
Ashish
